src/fact_checker/mod.rs

```diff
@@ -0,0 +1,34 @@
+pub mod error;
+pub mod fact_info;
+pub mod fact_node;
+pub mod fact_topology;
+
+use alloy::primitives::{Address, B256};
+use alloy::providers::{ProviderBuilder, RootProvider};
+use alloy::transports::http::{Client, Http};
+use url::Url;
+
+use crate::contracts::FactRegistry::{self, FactRegistryInstance};
+
+use self::error::FactCheckerError;
+
+pub struct FactChecker {
+    fact_registry: FactRegistryInstance<TransportT, ProviderT>,
+}
+
+type TransportT = Http<Client>;
+type ProviderT = RootProvider<TransportT>;
+
+impl FactChecker {
+    pub fn new(rpc_node_url: Url, verifier_address: Address) -> Self {
+        let provider = ProviderBuilder::new().on_http(rpc_node_url);
```

Comment:
should we store the provider in the global config and pass it here? it might be used in other places as well

Reply:
This case is rather specific (read only access), typically we will need a more involved provider (in Alloy it is now a much more complicated abstraction)

Reply:
Oh interesting. But won't providers by default be read only and you need to attach them to an account to get write access?